### PR TITLE
DEV: Use the new component-based modal API

### PR DESCRIPTION
**Pull request by CvX**

### CODE REVIEW

File: assets/javascripts/discourse/components/modal/assign-user.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import I18n from "I18n";
+
+export default class AssignUser extends Component {
+  @service taskActions;
+
+  formApi = {
+    sumbit() {},
```
Comment (davidtaylorhq) [inline assets/javascripts/discourse/components/modal/assign-user.js:10]:
```suggestion
    submit() {},
```

File: assets/javascripts/discourse/components/modal/assign-user.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import Component from "@glimmer/component";
+import { action } from "@ember/object";
+import { inject as service } from "@ember/service";
+import I18n from "I18n";
+
+export default class AssignUser extends Component {
+  @service taskActions;
+
+  formApi = {
```
Comment (davidtaylorhq) [inline assets/javascripts/discourse/components/modal/assign-user.js:9]: Given this is a little unusual, I think adding a comment would be good

```suggestion
  // `submit` property will be mutated by the `AssignUserForm` component
  formApi = {
```

File: assets/javascripts/discourse/controllers/bulk-assign.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import Controller, { inject as controller } from "@ember/controller";
+import { action } from "@ember/object";
+
+export default class BulkAssign extends Controller {
+  @controller topicBulkActions;
+
+  model;
+  formApi = {
```
Comment (davidtaylorhq) [inline assets/javascripts/discourse/controllers/bulk-assign.js:8]:
```suggestion

  // `submit` property will be mutated by the `AssignUserForm` component
  formApi = {
```

File: assets/javascripts/discourse/templates/modal/bulk-assign.hbs [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<div>
+  <AssignUserForm
+    @model={{this.model}}
+    @closeModal={{this.closeModal}}
```
Comment (davidtaylorhq) [inline assets/javascripts/discourse/templates/modal/bulk-assign.hbs:4]: Is this used? I don't see any reference to `closeModal` in the `AssignUserForm` implementation 🤔

File: assets/javascripts/discourse/components/modal/assign-user.js (hunk 1 above)
Comment (CvX) [reply assets/javascripts/discourse/components/modal/assign-user.js:10]: me, reading/watching absolutely anything: oh, here's a typo
me, reading this suggestion: why are these lines identical

🤣

**Review by davidtaylorhq — APPROVED** [review]

**Pull request merged**